mqtt/mqttc/Make.defs

```diff
@@ -0,0 +1,32 @@
+############################################################################
+# apps/mqtt/mqttc/Make.defs
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.  The
+# ASF licenses this file to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance with the
+# License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+# License for the specific language governing permissions and limitations
+# under the License.
+#
+############################################################################
+
+ifneq ($(CONFIG_MQTT_MQTTC),)
+CONFIGURED_APPS += $(APPDIR)/mqtt/mqttc
+
+CFLAGS   += -D__unix__
+CXXFLAGS += -D__unix__
```

Comment:
Should we contribute `__NuttX__` to the upstream library instead?  We've got similar situations with other 3rd party software as well. [mbedTLS](https://github.com/apache/incubator-nuttx-apps/pull/566/files#diff-6043c39303f7b14be00e75ffcac170a85c3ee0b7369e7db85f4131ef85280c81R39) is one of them.

Reply:
Yeah, I think it is a simple change to request. In the meantime we can leave this definition.

Reply:
```__unix__``` is only used in implementation, it's better to move CLFAGS/CXXFLAGS into Makefile

Reply:
@Ouss4 how about this comment?

Reply:
@xiaoxiang781216 tried to move them (as is done in other places) but they don't seem to take any effect in Makefile.  Any thing to look for here?

Reply:
I just create a patch to mainline: https://github.com/LiamBindle/MQTT-C/pull/131
We can remove these lines once the patch get merged, let's ignore it now.

Reply:
But we'll still need to wait for the next release.  Or use master?